src/core/electrostatics/p3m.cpp

```diff
@@ -366,24 +371,28 @@ template <int cao> struct AssignForces {
                                        p3m.E_mesh[2][ind]};
         });
 
-        p.force() -= pref * force;
+        p_force -= pref * force;
         ++p_index;
       }
     }
   }
 };
 
-auto dipole_moment(Particle const &p, BoxGeometry const &box) {
-  return p.q() * unfolded_position(p.pos(), p.image_box(), box.length());
+auto dipole_moment(double const q, Utils::Vector3d const unfolded_pos) {
```

Comment:
```suggestion
static auto dipole_moment(double const q, Utils::Vector3d const &unfolded_pos) {
```

Reply:
The `static` keyword is redundant here since we are in an anonymous namespace. Alternatively we can also replace the anonymous namespace with a `detail` namespace as above

Reply:
I have already replaced the anonymous namespaces by `static` functions to make it more explicit in the PR that removes global variables ([diff](https://github.com/espressomd/espresso/pull/4783/commits/11e1df1c9d6d3639a249d6bc6e129d65f382ae66#diff-c72eb42c3b85ab12f7d6b7b86dfa8d29ee3aa84fd4fa190f5ce6012e5a5740a9R386)).

Reply:
Updating when #4783 is merged.

Reply:
This is still not resolved: the `static` is missing (we're no longer in an anonymous namespace after the rebase), and we want to avoid a pass-by-copy.